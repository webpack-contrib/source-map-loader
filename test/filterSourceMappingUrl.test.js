import {
  compile,
  getCodeFromBundle,
  getCompiler,
  getErrors,
  normalizeMap,
  getWarnings,
} from "./helpers";

describe("filterSourceMappingUrl option", () => {
  it("should work", async () => {
    expect.assertions(6);

    const testId = "external-source-map.js";
    const compiler = getCompiler(testId, {
      filterSourceMappingUrl: (sourceMappingURL, resourcePath) => {
        expect(sourceMappingURL).toBeDefined();
        expect(resourcePath).toBeDefined();

        return true;
      },
    });
    const stats = await compile(compiler);
    const codeFromBundle = getCodeFromBundle(stats, compiler);

    expect(codeFromBundle.css).toMatchSnapshot("css");
    expect(normalizeMap(codeFromBundle.map)).toMatchSnapshot("map");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });

  it("should work with true value", async () => {
    const testId = "external-source-map.js";
    const compiler = getCompiler(testId, {
      filterSourceMappingUrl: () => true,
    });
    const stats = await compile(compiler);
    const codeFromBundle = getCodeFromBundle(stats, compiler);

    expect(codeFromBundle.css).toMatchSnapshot("css");
    expect(normalizeMap(codeFromBundle.map)).toMatchSnapshot("map");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });

  it('should work with "skip" value', async () => {
    const testId = "http-source-map.js";
    const compiler = getCompiler(testId, {
      filterSourceMappingUrl: () => "skip",
    });
    const stats = await compile(compiler);
    const codeFromBundle = getCodeFromBundle(stats, compiler);

    expect(codeFromBundle.map).toBeUndefined();
    expect(codeFromBundle.css).toMatchSnapshot("css");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });

  it('should work with "remove" value', async () => {
    const testId = "http-source-map.js";
    const compiler = getCompiler(testId, {
      filterSourceMappingUrl: () => "remove",
    });
    const stats = await compile(compiler);
    const codeFromBundle = getCodeFromBundle(stats, compiler);

    expect(codeFromBundle.map).toBeUndefined();
    expect(codeFromBundle.css).toMatchSnapshot("css");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });

  it("should work with false value", async () => {
    const testId = "http-source-map.js";
    const compiler = getCompiler(testId, {
      filterSourceMappingUrl: () => false,
    });
    const stats = await compile(compiler);
    const codeFromBundle = getCodeFromBundle(stats, compiler);

    expect(codeFromBundle.map).toBeUndefined();
    expect(codeFromBundle.css).toMatchSnapshot("css");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });

  it("should emit error", async () => {
    const testId = "external-source-map.js";
    const compiler = getCompiler(testId, {
      filterSourceMappingUrl: () => {
        throw new Error("error");
      },
    });
    const stats = await compile(compiler);

    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });
});
