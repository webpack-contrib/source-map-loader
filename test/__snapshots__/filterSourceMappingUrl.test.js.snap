// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filterSourceMappingUrl option should emit error: errors 1`] = `
Array [
  "ModuleBuildError: Module build failed (from \`replaced original path\`):
Error: error",
]
`;

exports[`filterSourceMappingUrl option should emit error: warnings 1`] = `Array []`;

exports[`filterSourceMappingUrl option should work with "remove" value: css 1`] = `
"with SourceMap
// comment
"
`;

exports[`filterSourceMappingUrl option should work with "remove" value: errors 1`] = `Array []`;

exports[`filterSourceMappingUrl option should work with "remove" value: warnings 1`] = `Array []`;

exports[`filterSourceMappingUrl option should work with "skip" value: css 1`] = `
"with SourceMap
//#sourceMappingURL=http://sampledomain.com/external-source-map2.map
// comment
"
`;

exports[`filterSourceMappingUrl option should work with "skip" value: errors 1`] = `Array []`;

exports[`filterSourceMappingUrl option should work with "skip" value: warnings 1`] = `Array []`;

exports[`filterSourceMappingUrl option should work with false value: css 1`] = `
"with SourceMap
// comment
"
`;

exports[`filterSourceMappingUrl option should work with false value: errors 1`] = `Array []`;

exports[`filterSourceMappingUrl option should work with false value: warnings 1`] = `Array []`;

exports[`filterSourceMappingUrl option should work with true value: css 1`] = `
"with SourceMap
// comment"
`;

exports[`filterSourceMappingUrl option should work with true value: errors 1`] = `Array []`;

exports[`filterSourceMappingUrl option should work with true value: map 1`] = `
Object {
  "file": "external-source-map.js",
  "mappings": "AAAA",
  "sources": Array [
    "/test/fixtures/external-source-map.txt - (normalized for test)",
  ],
  "sourcesContent": Array [
    "with SourceMap",
  ],
  "version": 3,
}
`;

exports[`filterSourceMappingUrl option should work with true value: warnings 1`] = `Array []`;

exports[`filterSourceMappingUrl option should work: css 1`] = `
"with SourceMap
// comment"
`;

exports[`filterSourceMappingUrl option should work: errors 1`] = `Array []`;

exports[`filterSourceMappingUrl option should work: map 1`] = `
Object {
  "file": "external-source-map.js",
  "mappings": "AAAA",
  "sources": Array [
    "/test/fixtures/external-source-map.txt - (normalized for test)",
  ],
  "sourcesContent": Array [
    "with SourceMap",
  ],
  "version": 3,
}
`;

exports[`filterSourceMappingUrl option should work: warnings 1`] = `Array []`;
